File: 892. Surface Area of 3D Shapes/surfaceArea.java
class surfaceArea {
    public int solution(int[][] grid) {
        int n = grid.length;
        int count = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (grid[i][j] > 0) {
                    int num = grid[i][j];
                    count = count + num * 4 + 2;
                    if (i > 0 && grid[i-1][j] > 0){
                        count = (num < grid[i-1][j]) ? (count - 2 * num) : (count - 2 * grid[i-1][j]);
                    }
                    if (j > 0 && grid[i][j-1] > 0){
                        count = (num < grid[i][j-1]) ? (count - 2 * num) : (count - 2 * grid[i][j-1]);
                    }
                }
            }
        }
        return count;
    }

    public static void main(String[] argv){
        surfaceArea obj = new surfaceArea();
        int[][] arr = {{2,2,2},{2,1,2},{2,2,2}};
        int res = obj.solution(arr);
        System.out.println(res);
    }
}

File: 892. Surface Area of 3D Shapes/surfaceArea.py
class surfaceArea:
    def solution(self, grid):
        N = len(grid)
        count = 0
        for i in range(0, N):
            for j in range(0, N):
                if grid[i][j] > 0:
                    count = count + 4 * grid[i][j] + 2
                    if i > 0 and grid[i-1][j] > 0:
                        count = count - 2 * grid[i][j] if grid[i][j] < grid[i-1][j] else count - 2 * grid[i-1][j]
                    if j > 0 and grid[i][j-1] > 0:
                        count = count - 2 * grid[i][j] if grid[i][j] < grid[i][j-1] else count - 2 * grid[i][j-1]
        return count

#grid = [[2]]
#grid = [[1,2],[3,4]]
#grid = [[1,0],[0,2]]
#grid = [[1,1,1],[1,0,1],[1,1,1]]
grid = [[2,2,2],[2,1,2],[2,2,2]]
obj = surfaceArea()
res = obj.solution(grid)
print(res)
